kubernetes/cache/namespaces.go

```diff
@@ -18,10 +18,16 @@ type (
 func (c *kialiCacheImpl) SetNamespaces(token string, namespaces []models.Namespace) {
 	defer c.tokenLock.Unlock()
 	c.tokenLock.Lock()
-	nameNamespace := make(map[string]models.Namespace, len(namespaces))
+	nameNamespace := make(map[string]map[string]models.Namespace)
 	for _, ns := range namespaces {
-		nameNamespace[ns.Name] = ns
+		for _, cluster := range ns.Clusters {
+			if nameNamespace[ns.Name] == nil {
+				nameNamespace[ns.Name] = make(map[string]models.Namespace)
+			}
+			nameNamespace[ns.Name][cluster] = ns
```

Comment:
For what I understand here, maybe I'm missing something, but for multiple clusters, the same namespace is stored in the map?.

I guess we need to store the namespace from the cluster because probably labels and annotations may vary between namespaces (with same name) across different clusters.

Reply:
This was something I forgot to change, but in a first approach I combined the namespace information and the namespace model was saving an array of namespaces where it lived, so this just reflects that situation. But you made a good point, as this needs to be updated.